### PR TITLE
find gcd of two numbers.

### PR DESCRIPTION
**Pull request by varuogm**
this is program for finding GCD of two numbers

#### Checklist

- [x ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x ] Relevant documentation/comments is changed or added
- [x ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x ] I acknowledge that all my contributions will be made under the project's license.

Notes:  program for GCD and explanation

### CODE REVIEW

File: others/GCD_program.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+/*C++ inbuilt function for finding GCD*/
+
+#include<iostream>
+using namespace std;
```
Comment (Panquesito7) [inline others/GCD_program.cpp:5]: Please remove this and use `std::` when necessary.

File: others/GCD_program.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
+/*C++ inbuilt function for finding GCD*/
+
+#include<iostream>
```
Comment (Panquesito7) [inline others/GCD_program.cpp:4]:
```suggestion
#include <iostream>
```

File: others/GCD_program.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
```
Comment (Panquesito7) [inline others/GCD_program.cpp:1]: Please add a comment structure like in #807.

File: others/GCD_program.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
+
+/*C++ inbuilt function for finding GCD*/
+
+#include<iostream>
+using namespace std;
+int gcd(int a, int b) 
+{
```
Comment (Panquesito7) [inline others/GCD_program.cpp:7]:
```suggestion
int gcd(int a, int b) {
```

File: others/GCD_program.cpp [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+#include<iostream>
+using namespace std;
+int gcd(int a, int b) 
+{ 
+	if (b == 0) 
+		return a; 
+	return gcd(b, a % b); 	
+} 
+int main() 
+{
```
Comment (Panquesito7) [inline others/GCD_program.cpp:13]:
```suggestion
int main() {
```

File: others/GCD_program.cpp [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+#include<iostream>
+using namespace std;
+int gcd(int a, int b) 
+{ 
+	if (b == 0) 
+		return a; 
+	return gcd(b, a % b); 	
+} 
+int main()
```
Comment (Panquesito7) [inline others/GCD_program.cpp:12]: Please add a comment structure like:

```c++
/**
 * Main function
 */
```

File: others/GCD_program.cpp [hunk 7]
```diff
@@ -0,0 +1,19 @@
+
+/*C++ inbuilt function for finding GCD*/
+
+#include<iostream>
+using namespace std;
+int gcd(int a, int b) 
+{ 
+	if (b == 0) 
+		return a; 
+	return gcd(b, a % b);
```
Comment (Panquesito7) [inline others/GCD_program.cpp:10]:
```suggestion
	if (b == 0) {
		return a;
	}
	return gcd(b, a % b);
```

File: others/GCD_program.cpp [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+	return gcd(b, a % b); 	
+} 
+int main() 
+{ 
+	int a = 8, b = 6; 
+	cout<<"GCD of two num is "<<gcd(a, b); 
+	return 0; 
+} 
+//also theere is simple way to do gcd by using    __gcd(m, n)  where m ,n are numbers
+
```
Comment (Panquesito7) [inline others/GCD_program.cpp:19]: Please remove this empty line.

File: others/GCD_program.cpp [hunk 9]
```diff
@@ -0,0 +1,19 @@
+
+/*C++ inbuilt function for finding GCD*/
+
+#include<iostream>
+using namespace std;
+int gcd(int a, int b)
```
Comment (Panquesito7) [inline others/GCD_program.cpp:6]: Undocumented function. Please document it.

**Comment by kvedala** [comment]
1. please review existing code - there already are 2 types of implementations for GCD
2. Are you contributing something new apart from them? the ensure you have read the contribution guidelines in the README of the repo.
3. Are you improving existing code? If so, ensure the file names are correct.

File: others/GCD_program.cpp [hunk 10]
```diff
@@ -9,6 +10,7 @@ int gcd(int a, int b) {
 	return gcd(b, a % b);
 } 
 int main() {
+//given two numbers
```
Comment (Panquesito7) [inline others/GCD_program.cpp:13]: Missing indentation.

File: others/GCD_program.cpp [hunk 11]
```diff
@@ -0,0 +1,17 @@
+/** @file
+ * C++ inbuilt function for finding GCD
+ */
+#include <iostream>
+int gcd(int a, int b) {
+	if (b == 0) {
```
Comment (mishraabhinn) [inline others/GCD_program.cpp:6]:
```
if(a<b){
return  gcd(b,a);
}
```
add this before check `(b==0)`

**Comment by kvedala** [comment]
> * please review existing code - there already are 2 types of implementations for GCD
> * Are you contributing something new apart from them? the ensure you have read the contribution guidelines in the README of the repo.
> * Are you improving existing code? If so, ensure the file names are correct.

Guys, until these questions are answered by the author, I think the reviews and fixes would be futile.

**Pull request closed without merging**